common/es/src/main/java/io/apiman/common/es/util/EsIndexMapping.java

```diff
@@ -26,249 +28,441 @@
+     * Sets Ip Type to index field
+     *
+     * @param fieldName            the field name
+     * @param indexFieldProperties the index field properties
+     */
+    private static void setIpTypeToIndexField(String fieldName, Map<String, Object> indexFieldProperties) {
+        // create keyword type property
+        Map<String, Object> keywordTypeProperty = new HashMap<>();
+        keywordTypeProperty.put("type", EsConstants.ES_MAPPING_TYPE_IP);
+        setFieldMapping(fieldName, keywordTypeProperty, indexFieldProperties);
+    }
+
+    /**
+     * Sets Long Type to index field
+     *
+     * @param fieldName            the field name
+     * @param indexFieldProperties the index field properties
+     */
+    private static void setLongTypeToIndexField(String fieldName, Map<String, Object> indexFieldProperties) {
+        // create keyword type property
+        Map<String, Object> keywordTypeProperty = new HashMap<>();
+        keywordTypeProperty.put("type", EsConstants.ES_MAPPING_TYPE_LONG);
+        setFieldMapping(fieldName, keywordTypeProperty, indexFieldProperties);
+    }
+
+    /**
+     * Sets Object Type to index field
+     *
+     * @param fieldName            the field name
      * @param indexFieldProperties the index field properties
      */
-    private static void addIpTypeToIndexField(String fieldName, Map<String, Object> indexFieldProperties) {
+    private static void setObjectTypeToIndexField(String fieldName, Map<String, Object> indexFieldProperties) {
         // create keyword type property
         Map<String, Object> keywordTypeProperty = new HashMap<>();
-        keywordTypeProperty.put("type", "ip");
-        addFieldMapping(fieldName, keywordTypeProperty, indexFieldProperties);
+        keywordTypeProperty.put("type", EsConstants.ES_MAPPING_TYPE_OBJECT);
+        setFieldMapping(fieldName, keywordTypeProperty, indexFieldProperties);
     }
 
     /**
      * Set field data to index text field
-     * @param fieldName the field name
+     *
+     * @param fieldName            the field name
      * @param indexFieldProperties the index field properties
      */
-    private static void setFieldDataToIndexTextField(String fieldName, Map<String, Object> indexFieldProperties) {
+    private static void setTextTypeToIndexField(String fieldName, Map<String, Object> indexFieldProperties) {
         // create keyword type property
         Map<String, Object> textTypeProperty = new HashMap<>();
-        textTypeProperty.put("type", "text");
+        Map<String, Object> keywordTypeProperty = new HashMap<>();
+        Map<String, Object> multiFieldProperty = new HashMap<>();
+
+        textTypeProperty.put("type", EsConstants.ES_MAPPING_TYPE_TEXT);
         textTypeProperty.put("fielddata", true);
-        addFieldMapping(fieldName, textTypeProperty, indexFieldProperties);
+        keywordTypeProperty.put("type", EsConstants.ES_MAPPING_TYPE_KEYWORD);
+        keywordTypeProperty.put("ignore_above", 256);
```

Comment:
@BastianGem Are these lines the default values (e.g. the 256)?

Reply:
@volkflo Yes these are deafult values.
The ignore_above property avoids indexing of huge terms, with the current setting of 256 only strings with a maximum length of 256 will be indexed